Integrations/EDL/EDL.py

```diff
@@ -0,0 +1,371 @@
+    """
+    Create a dictionary for output values using the selected format
+    """
+    out_format_func = {
+        FORMAT_TEXT: format_ioc_text_out,
+        FORMAT_JSON_SEQ: format_ioc_json_seq_out,
+        FORMAT_CSV: format_ioc_csv_out
+    }
+    return create_formatted_values_out_dict(iocs, out_format, out_format_func.get(out_format, str))
+
+
+def create_formatted_values_out_dict(iocs: list, out_format: str, out_format_func: Callable) -> dict:
+    """
+    Create a dictionary for output values formatted in the selected out_format
+    """
+    out_dict = {}
+    if out_format == FORMAT_JSON:
+        iocs_list = [ioc for ioc in iocs]
+        return {'iocs_list': json.dumps(iocs_list, indent=4)}
+    else:
+        for ioc in iocs:
+            value = ioc.get('value')
+            if value:
+                out_dict[value] = out_format_func(ioc)
+        if out_format == 'csv' and len(iocs) > 0:
+            # add csv headers to out_dict under saved key
+            headers = list(iocs[0].keys())
+            out_dict[CSV_FIRST_LINE_KEY] = list_to_str(headers)
+        return out_dict
+
+
+def format_ioc_text_out(ioc: dict) -> str:
+    """
+    Return output in text format
+    """
+    return ioc.get('value', '')
+
+
+def format_ioc_json_seq_out(ioc: dict) -> str:
+    """
+    Return output in json seq format
+    """
+    return json.dumps(ioc)
+
+
+def format_ioc_csv_out(ioc: dict) -> str:
+    """
+    Return output in csv format
+    """
+    values = list(ioc.values())
+    # wrap csv values with " to separate them
+    return list_to_str(values, map_func=lambda val: f'"{val}"')
+
+
+def get_edl_ioc_values_list() -> list:
+    """
+    Get the ioc list to return in the edl
+    """
+    params = demisto.params()
+    out_format = params.get('format')
+    on_demand = params.get('on_demand')
+    limit = parse_integer(params.get('edl_size'), EDL_LIMIT_ERR_MSG)
+    # on_demand ignores cache
+    if on_demand:
+        values = get_ioc_values_list_from_context(out_format)
+    else:
+        last_run = demisto.getLastRun().get('last_run')
+        indicator_query = demisto.params().get('indicators_query')
+        if last_run:
+            cache_refresh_rate = demisto.params().get('cache_refresh_rate')
+            cache_time, _ = parse_date_range(cache_refresh_rate, to_timestamp=True)
+            if last_run <= cache_time:
+                values = refresh_edl_context(indicator_query, out_format, limit=limit)
+            else:
+                values = get_ioc_values_list_from_context(out_format)
+        else:
+            values = refresh_edl_context(indicator_query, out_format, limit=limit)
+    return values
+
+
+def get_ioc_values_list_from_context(out_format: str) -> list:
+    """
+    Extracts output values from cache
+    """
+    cache_dict = demisto.getIntegrationContext()
+    values = create_csv_out_list(cache_dict) if out_format == FORMAT_CSV else list(cache_dict.values())
```

Comment:
this will work also with text format?

Reply:
yeah, but this is irrelevant as I changed the way the data is stored, so no need to implement diff logic in the extract